website/docs/r/app_service.html.markdown

```diff
@@ -172,6 +214,81 @@ A `cors` block supports the following:
 ---
 
 Elements of `ip_restriction` support:
+`auth_settings` supports the following:
+
+* `enabled` - (Required) Is Authentication enabled?
+
+* `additional_login_params` - (Optional) Login parameters to send to the OpenID Connect authorization endpoint when a user logs in. Each parameter must be in the form "key=value".
+
+* `allowed_external_redirect_urls` - (Optional) External URLs that can be redirected to as part of logging in or logging out of the app.
+
+~> **NOTE:** When providing multiple providers, the default provider must be set for settings like `unauthenticated_client_action` to work. 
+
+* `default_provider` - (Optional) The default provider to use when multiple providers have been set up. Possible values are `AzureActiveDirectory`, `Facebook`, `Google`, `MicrosoftAccount` and `Twitter`.
+
+* `issuer` - (Optional) Issuer URI. When using Azure Active Directory, this value is the URI of the directory tenant, e.g. https://sts.windows.net/{tenant-guid}/.
+
+* `runtime_version` - (Optional) The runtime version of the Authentication/Authorization module.
+
+* `token_refresh_extension_hours` - (Optional) The number of hours after session token expiration that a session token can be used to call the token refresh API. Defaults to 72.
+
+* `token_store_enabled` - (Optional) If enabled the module will durably store platform-specific security tokens that are obtained during login flows. Defaults to false.
+
+* `unauthenticated_client_action` - (Optional) The action to take when an unauthenticated client attempts to access the app. Possible values are `AllowAnonymous` and `RedirectToLoginPage`.
+
+* `active_directory` - (Optional) Active directory Authentication configuration.
+
+* `facebook` - (Optional) Facebook Authentication configuration.
+
+* `google` - (Optional) Google Authentication configuration.
+
+* `microsoft` - (Optional) Microsoft Account Authentication configuration.
+
+* `twitter` - (Optional) Twitter Authentication configuration.
```

Comment:
for consistency with other resources can we make this:

```suggestion
* `twitter` - (Optional) A `twitter` block as defined below.
```

(_minor_ could we also update the ordering so these are sorted as Required -> Optional and then Alphabetically?)